### PR TITLE
color.cpp split

### PR DESCRIPTION
**Pull request by savuor**
### This pullrequest changes

* color.cpp and cvtcolor.cl are split to several files by color spaces families
* each color conversion has its own interface

### To be discussed

* new functions interfaces and what of them are to be exposed to user

### CODE REVIEW

**Comment by mshabunin** [comment]
GCC-7.2.0 produces a warning:
```
In file included from opencv/modules/imgproc/src/color_lab.cpp:11:0:
opencv/modules/imgproc/src/color.hpp:432:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
 #if NEED_IPP
     ^~~~~~~~
```
Maybe it will be better to use more conventional construction:
```.cpp
#if defined (HAVE_IPP) && (IPP_VERSION_X100 >= 700)
#  define NEED_IPP 1
#else
#  define NEED_IPP 0
#endif
```

**Comment by savuor** [comment]
@mshabunin done as you suggested, hope this will fix the warning

File: modules/imgproc/src/color.hpp [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+        default:
+            dstSz = sz;
+            break;
+        }
+
+        _dst.create(dstSz, CV_MAKETYPE(depth, dcn));
+        dst = _dst.getUMat();
+
+        srcarg = ocl::KernelArg::ReadOnlyNoSize(src);
+        dstarg = ocl::KernelArg::WriteOnly(dst);
```
Comment (alalek) [inline modules/imgproc/src/color.hpp]: It is not necessary to store OpenCL "srcarg/dstarg" in fields of this class.

File: modules/imgproc/src/color.hpp [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+        for( int i = range.start; i < range.end; ++i, yS += src_step, yD += dst_step )
+            cvt(reinterpret_cast<const _Tp*>(yS), reinterpret_cast<_Tp*>(yD), width);
+    }
+
+private:
+    const uchar * src_data;
+    size_t src_step;
+    uchar * dst_data;
+    size_t dst_step;
+    int width;
```
Comment (alalek) [inline modules/imgproc/src/color.hpp]: "const" for these fields

File: modules/imgproc/src/color.hpp [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+    {
+    case COLOR_Lab2BGR: case COLOR_Lab2RGB: case COLOR_Lab2LBGR: case COLOR_Lab2LRGB:
+    case COLOR_BGR2Lab: case COLOR_RGB2Lab: case COLOR_LBGR2Lab: case COLOR_LRGB2Lab:
+        return true;
+    default:
+        return false;
+    }
+}
+
+inline bool issRGB(int code)
```
[the remark this answers is not in the document]
Comment (savuor) [reply modules/imgproc/src/color.hpp]: It's not a typo, sRGB is a color space
But I agree that the name is confusing, `isSRGB` would be better

**Comment by savuor** [comment]
@alalek fixes provided based on your comments

File: modules/imgproc/src/color.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (alalek) [reply modules/imgproc/src/color.hpp]: `is_sRGB` should be fine

**Pull request merged**